master/internal/command/command.go

```diff
@@ -281,11 +268,71 @@ func (c *command) handleAPIRequest(ctx *actor.Context, apiCtx echo.Context) {
 	}
 }
 
+func (c *command) receiveSchedulerMsg(ctx *actor.Context) error {
+	switch msg := ctx.Message().(type) {
+	case scheduler.ResourcesAllocated:
+		// Ignore this message if the command has exited.
+		if c.task == nil || msg.ID != c.task.ID {
+			ctx.Log().Info("ignoring resource allocation since the command has exited.")
+			return nil
+		}
+
+		check.Panic(check.Equal(len(msg.Allocations), 1,
```

Comment:
I think here is a good place to use panic since it should be definitely our fault if more than one allocations are sent to the command and we can catch them early by using panic here. Reference: https://gobyexample.com/panic#:~:text=Go%20by%20Example%3A%20Panic,-A%20panic%20typically&text=This%20is%20the%20only%20program%20on%20the%20site%20designed%20to%20panic.&text=A%20common%20use%20of%20panic,when%20creating%20a%20new%20file.

Reply:
In our system we can return an error or panic here, unfortunately we do not have consistency although in the majority of cases we do return errors instead of panics which is why I suggested we make this change.

Reply:
talked offline. Generally, we should be consistent, which we aren't. According to the convention of using errors and panics based on whether they are expected or caused by users' wrong input, we should use panic here.